AfterClass-io/afterclass.io-v2
# feat: validate school domain for user sign up

closes #11

## Changes
- add school domains as part of environment variables
- add email validation function under `src/common/functions`
- add email checking in zod email validation on client and server side
- add redirection to `verify` page after sign up is complete, and before email is verified
- display error message if registered user tries to register again

## Implementation
1. We have decided to use a single string to represent the school's email domain rather than an array. Our research indicates that the domain name after the '@' symbol in email addresses remains consistent, regardless of any prefixes. See [SMU example.](https://itsupport.smu.edu.sg/hc/en-us/articles/204146070-Domain-Name-Management-Service#:~:text=SMU%20is%20the%20owner%20of,smu.edu.sg%E2%80%9D.)
2. If registered user try to sign up again, supabase api will throw an error, which is caught by the program and error message will be displayed (see [supabase create user api](https://supabase.com/docs/reference/javascript/auth-signup))

## Testing
To set up the project locally for testing, follow the following steps:
1. Install dependencies
`yarn install`
3. Run the development server
` yarn dev`
4. Head over to user sign up page
5. Fill in valid school emails and password
6. If account creation is successful, you should be redirected to email verification page

## Preview
![Auth](https://github.com/AfterClass-io/afterclass.io-v2/assets/68802506/49c15c8d-b7aa-4496-837f-cb2a77312e7f)
